decompiler_agent/src/main/java/org/jrd/agent/api/Variables.java

```diff
@@ -0,0 +1,95 @@
+package org.jrd.agent.api;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Variables {
+
+    /**
+     * we have to have all inernal classes inited, so any later classes can use them without manual INIT.
+     */
+    public static void init() {
+        NoSuchFakeFieldException.init();
+        Global.init();
+    }
+
+    protected Variables() {
+        // prevents calls from subclass... checkstile sucks
+        throw new UnsupportedOperationException();
+    }
+
+    public static class NoSuchFakeFieldException extends RuntimeException {
```

Comment:
FakeField is such a weird naming scheme. We're in the Variables class, why not have a `NoSuchVariableException`?

Reply:
renamed